ui/components/app/confirm-data/index.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+
+import { getKnownMethodData } from '../../../selectors';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { useTransactionFunctionType } from '../../../hooks/useTransactionFunctionType';
+
+import Disclosure from '../../ui/disclosure';
+import TransactionDecoding from '../transaction-decoding';
+
+const ConfirmData = ({ txData, dataComponent }) => {
+  const t = useI18nContext();
+  const { txParams = {} } = txData;
+  const methodData = useSelector(
+    (state) => getKnownMethodData(state, txParams.data) || {},
+  );
+  const { functionType } = useTransactionFunctionType(txData);
+
+  if (dataComponent) {
+    return dataComponent;
+  }
+
+  if (!txParams.data) {
+    return null;
+  }
+
+  const { params } = methodData;
+  const functionParams = params?.length
+    ? `(${params.map(({ type }) => type).join(', ')})`
+    : '';
+
+  return (
+    <div className="confirm-page-container-content__data">
```

Comment:
REQ: now is the time to move these styles from where they are, and adopt the Box component instead of the divs.

Reply:
We should try to co-locate styles wherever possible and avoid styles that can be achieved with our design system components.